### PR TITLE
enforce must always have at least one owner

### PR DESCRIPTION
**Pull request by wilsoncusack**
Adds a guard on removeOwnerAtIndex such that it will revert if the call will leave no owners on the account. 

No tests yet because still considering. Removes the ability to renounce ownership, except by setting some like 0xdead owner.

### CODE REVIEW

File: src/MultiOwnable.sol [hunk 1]
```diff
@@ -82,31 +92,47 @@ contract MultiOwnable {
     /// @notice Convenience function to add a new owner address.
     ///
     /// @param owner The owner address.
-    function addOwnerAddress(address owner) public virtual onlyOwner {
+    function addOwnerAddress(address owner) external virtual onlyOwner {
```
Comment (wilsoncusack) [inline src/MultiOwnable.sol:95]: other clean up I am grouping in here

File: src/MultiOwnable.sol [hunk 2]
```diff
@@ -189,12 +222,30 @@ contract MultiOwnable {
     function _addOwnerAtIndex(bytes memory owner, uint256 index) internal virtual {
         if (isOwnerBytes(owner)) revert AlreadyOwner(owner);
 
-        _getMultiOwnableStorage().isOwner[owner] = true;
-        _getMultiOwnableStorage().ownerAtIndex[index] = owner;
+        MultiOwnableStorage storage $ = _getMultiOwnableStorage();
+        $.isOwner[owner] = true;
```
Comment (wilsoncusack) [inline src/MultiOwnable.sol:226]: G-03 https://github.com/code-423n4/2024-03-coinbase-findings/issues/195

File: src/MultiOwnable.sol [hunk 3]
```diff
@@ -189,12 +222,30 @@ contract MultiOwnable {
     function _addOwnerAtIndex(bytes memory owner, uint256 index) internal virtual {
         if (isOwnerBytes(owner)) revert AlreadyOwner(owner);
 
-        _getMultiOwnableStorage().isOwner[owner] = true;
-        _getMultiOwnableStorage().ownerAtIndex[index] = owner;
+        MultiOwnableStorage storage $ = _getMultiOwnableStorage();
```
Comment (wilsoncusack) [inline src/MultiOwnable.sol:225]: G-03 https://github.com/code-423n4/2024-03-coinbase-findings/issues/195

File: src/MultiOwnable.sol [hunk 4]
```diff
@@ -52,8 +54,9 @@ contract MultiOwnable {
     /// @notice Thrown when `owner` argument does not match owner found at index.
     ///
     /// @param index The index of the owner to be removed.
-    /// @param owner Owner to be removed which did not match owner found at index.
-    error WrongOwnerAtIndex(uint256 index, bytes owner);
+    /// @param expectedOwner The owner passed in the remove call.
+    /// @param actualOwner The actual owner at `index`.
+    error WrongOwnerAtIndex(uint256 index, bytes expectedOwner, bytes actualOwner);
```
Comment (wilsoncusack) [inline src/MultiOwnable.sol:59]: @stevieraykatz this felt nice to have to me

Comment (stevieraykatz) [reply src/MultiOwnable.sol:59]: Agreed -- could help track down malicious replays

File: src/MultiOwnable.sol (hunk 1 above)
Comment (stevieraykatz) [reply src/MultiOwnable.sol:102]: I'm wondering if this is overly restrictive. I could imagine a plugin/module that might help users manage owners. Depending on how the community agrees on module/plugin support, this call could be executed in the context of  this contract.

Comment (stevieraykatz) [reply src/MultiOwnable.sol:102]: Fair point

**Review by stevieraykatz — APPROVED** [review]

File: src/MultiOwnable.sol (hunk 3 above)
Comment (UUndead) [reply src/MultiOwnable.sol:236]: Wt ru guys doing

File: src/MultiOwnable.sol (hunk 1 above)
Comment (UUndead) [reply src/MultiOwnable.sol:102]: Wt now? DM me pls

**Pull request merged**